### PR TITLE
Content mgmt update

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: None

## Description:
- The `MarketplacePackInstaller` script can now accept packs' data to install as an argument.
- The `CustomPackInstaller` script now fails the task when it gets a bad installation status.
- Updated playbook accordingly.

### CODE REVIEW

**Comment by yaron-libman** [comment]
@ShirleyDenkberg please work with Shannon to get this done. Seems relatively small.

File: Packs/ContentManagement/Scripts/CustomPackInstaller/CustomPackInstaller.py [hunk 1]
```diff
@@ -65,6 +65,9 @@ def main():
             )
         )
 
+        if installation_status != 'Success':
```
Comment (yaakovi) [inline Packs/ContentManagement/Scripts/CustomPackInstaller/CustomPackInstaller.py:68]: do we have a reason for the failure?
where can the user get more information about it?

File: Packs/ContentManagement/Scripts/MarketplacePackInstaller/MarketplacePackInstaller.py [hunk 2]
```diff
@@ -237,22 +237,29 @@ def install_pack_and_its_dependencies_recursively(self, pack_data: Dict[str, str
...
+    try:
+        return [
+            {
+                'id': pack[id_key],
+                'version': pack[version_key],
+            }
+            for pack in packs_data  # type: ignore
+        ]
+    except KeyError as e:
+        raise DemistoException(f'The following key was does not exist in the packs data: {e}.') from e
```
Comment (yaakovi) [inline Packs/ContentManagement/Scripts/MarketplacePackInstaller/MarketplacePackInstaller.py:260]:
```suggestion
        raise DemistoException(f'The following key does not exist in the pack data: {e}.') from e
```

File: Packs/ContentManagement/Scripts/MarketplacePackInstaller/MarketplacePackInstaller.py [hunk 3]
```diff
@@ -237,22 +237,29 @@ def install_pack_and_its_dependencies_recursively(self, pack_data: Dict[str, str
...
+    id_key = args.get('pack_id_key')
+    version_key = args.get('pack_version_key')
+
+    try:
+        return [
+            {
+                'id': pack[id_key],
+                'version': pack[version_key],
+            }
+            for pack in packs_data  # type: ignore
```
Comment (yaakovi) [inline Packs/ContentManagement/Scripts/MarketplacePackInstaller/MarketplacePackInstaller.py:257]: why do you need "type ignore" here?

File: Packs/ContentManagement/Scripts/MarketplacePackInstaller/MarketplacePackInstaller.py [hunk 4]
```diff
@@ -237,22 +237,29 @@ def install_pack_and_its_dependencies_recursively(self, pack_data: Dict[str, str
...
-    """Fetched packs' data from context and formats it to an installable object.
+def format_packs_data_for_installation(args) -> List[Dict[str, str]]:
+    """Creates the body of the installation request from the raw data.
 
     Returns:
         List[Dict[str, str]]: Installable objects list.
     """
-    instance_context = demisto.context()
-    context_packs_data = instance_context.get('ConfigurationSetup', {}).get('MarketplacePacks')
+    packs_data = args.get('packs_data', [])
```
Comment (yaakovi) [inline Packs/ContentManagement/Scripts/MarketplacePackInstaller/MarketplacePackInstaller.py:246]: use `argToList`

**Comment by shannon-holland** [comment]
@yaakovi Document review completed.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**